### PR TITLE
feat(suite): add FW hash check whenever device connects

### PR DESCRIPTION
**Pull request by peter-sanderson**
Refactoring from this PR moved to https://github.com/trezor/trezor-suite/pull/13162

### CODE REVIEW

File: suite-common/wallet-core/src/firmware/firmwareThunks.ts [hunk 1]
```diff
@@ -47,17 +47,27 @@ export const checkFirmwareAuthenticity = createThunk(
...
+        const baseUrl = `${isDesktop() ? desktopBinDir : resolveStaticPath('connect/data')}${
+            useDevkit ? '/devkit' : ''
+        }`;
+
         const result = await TrezorConnect.checkFirmwareAuthenticity({
-            device: {
-                path: device.path,
-            },
+            device: { path: device.path },
+            baseUrl,
```
Comment (peter-sanderson) [inline suite-common/wallet-core/src/firmware/firmwareThunks.ts]: This way, if FW is bundled with Suite it wont be downloaded.

File: packages/connect/src/api/checkFirmwareAuthenticity.ts [hunk 2]
```diff
@@ -52,11 +54,18 @@ export default class CheckFirmwareAuthenticity extends AbstractMethod<'checkFirm
...
+        // const result = await this.device
+        //     .getCommands()
+        //     .typedCall('GetFirmwareHash', 'FirmwareHash', {
+        //         challenge,
+        //     });
+
+        const result: TypedResponseMessage<'FirmwareHash'> = {
+            type: 'FirmwareHash',
+            message: {
+                hash: 'yolo',
```
Comment (peter-sanderson) [inline packages/connect/src/api/checkFirmwareAuthenticity.ts]: Todo: remove this, this is just for testing

File: suite-common/wallet-core/src/firmware/checkFirmwareHashThunk.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+export const checkFirmwareHashAndFailSuiteIfInvalidThunk = createThunk<
+    void,
+    CheckFirmwareHashThunkParams,
+    { rejectValue: { error: string } }
+>(`${FIRMWARE_MODULE_PREFIX}/checkFirmwareHashThunk`, async ({ device }, { dispatch }) => {
+    const result = await dispatch(validateFirmwareHashThunk({ device }));
+
+    if (result.payload === undefined || result.payload.success !== 'valid') {
+        console.log('___FAAAAAIL', result.payload);
```
Comment (peter-sanderson) [inline suite-common/wallet-core/src/firmware/checkFirmwareHashThunk.ts:71]: todo: remove

File: packages/suite/src/hooks/suite/useFirmware.ts [hunk 4]
```diff
@@ -121,7 +116,6 @@ export const useFirmware = () => {
         originalDevice,
         firmwareUpdate: (...params: Parameters<typeof firmwareUpdate>) =>
             dispatch(firmwareUpdate(...params)),
-        checkFirmwareAuthenticity: () => dispatch(checkFirmwareAuthenticity()),
```
Comment (peter-sanderson) [inline packages/suite/src/hooks/suite/useFirmware.ts:124]: Absolutely not needed here. Can be dispatched directly

File: packages/suite/src/components/suite/PrerequisitesGuide/HashCheckFailed.tsx [hunk 5]
```diff
@@ -0,0 +1,6 @@
+// import { SecurityCheckFail } from '../../../views/onboarding/steps/SecurityCheck/SecurityCheckFail';
+
+export const HashCheckFailed = () => {
+    return <>Ba-da bum</>;
```
Comment (komret) [inline packages/suite/src/components/suite/PrerequisitesGuide/HashCheckFailed.tsx:4]: design needed before merging

File: suite-common/wallet-core/src/firmware/checkFirmwareHashThunk.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+        // FW binaries are stored in "*/static/connect/data/firmware/*/*.bin". see "connect-common" package
+        const baseUrl = `${isDesktop() ? desktopBinDir : resolveStaticPath('connect/data')}${
+            useDevkit ? '/devkit' : ''
+        }`;
+
+        const result = await TrezorConnect.checkFirmwareAuthenticity({
+            // For current version of the FW (the one known by the Suite) we dont need to download it
+            // and we just link it statically
+            baseUrl: eq(currentFwVersion, availableFwVersion) ? baseUrl : undefined,
```
Comment (komret) [inline suite-common/wallet-core/src/firmware/checkFirmwareHashThunk.ts:46]: Is this worth installing the lib? Looks like you could just do `===`.

**Comment by peter-sanderson** [comment]
Closed in favor of: https://github.com/trezor/trezor-suite/pull/14766

**Pull request closed without merging**